app/components/practice-history-chart.tsx

```diff
@@ -212,3 +215,12 @@ export function EchartsComponent(props: {
 
   return <div className={props.className} ref={React.useCallback(elRef, [])} />;
 }
+
+// quick-and-dirty tree-shaking code-spliting for echarts
+export function useEcharts() {
+  return useQuery({
+    ...usePromiseQueryOpitons(() =>
+      !import.meta.env.SSR ? import("echarts") : undefined!
+    ),
+  });
+}
```

Comment:
I guess dynamic importing `echarts` really doesn't matter as long as it's excluded from server bundle since all the pages which uses `echarts` doesn't nothing while waiting `echarts`.

Thus, for now, just re-exporting `echarts.init` from client only file should suffice.